### ISSUE TITLE
Host option set to a Hostname with both IPv4 and IPv6 addresses may not work 

### ISSUE THREAD

**Issue by yjiang-c** [fluent/fluent-bit#2838]
If I set Host option in an OUTPUT plugin to be a host name with both IPv4 and IPv6 addresses (DNS resolve return both IPv4 or IPv6 addresses), then fluentbit may not able to connect to the host name.

The issue is caused by separate locations to determine the IP version for creating socket and resolve IP addresses. In connecting to the target host, the following code create socket based on configure option ipv6 shown as below in flb_io_net_connect() function in file flb_io.c
  
```
        /* Create the socket */                                                     
        if (u_conn->u->flags & FLB_IO_IPV6) {                                       
            fd = flb_net_socket_create(AF_INET6, async);                            
        }                                                                           
        else {                                                                      
            fd = flb_net_socket_create(AF_INET, async);                             
        }                               
```

Then in the function flb_net_tcp_fd_connect() of file flb_network.c, it try to let system automatically choose a IP version of target host by specifying hints.ai_family = AF_UNSPEC;
```
    hints.ai_family = AF_UNSPEC;                                                 
    hints.ai_socktype = SOCK_STREAM;                                             
                                                                                 
    snprintf(_port, sizeof(_port), "%lu", port);                                 
    ret = getaddrinfo(host, _port, &hints, &res);                                
    if (ret != 0) {                                                              
        flb_warn("net_tcp_fd_connect: getaddrinfo(host='%s'): %s",               
                 host, gai_strerror(ret));                                       
        return -1;                                                               
    }                                                                            
                                                                                 
    ret = connect(fd, res->ai_addr, res->ai_addrlen); 
```

In this way, we may create IPv4 socket to connect to a IPv6 address. I encountered this issue on Ubuntu 18.04. 

The desired behaviour is to remove ipv6 option and let fluentbit smartly to choose IPv4 or IPv6 address by getaddrinfo(). The client socket created should depends on the result of getaddrinfo().

**Comment by edsiper** [comment]
@yjiang-c 

thanks for raising this issue.

Would you please check the code base of GIT master ?, some weeks ago this was modified and now it properly checks the returned address by DNS, so the old ```ipv6``` option will be deprecated.

ref: https://github.com/fluent/fluent-bit/blob/master/src/flb_network.c#L433

**Comment by yjiang-c** [comment]
Thanks for your quick response. I compiled the latest master branch and found the issue has been fixed. Do you know whether the fix is included in 1.6.8?

**Comment by edsiper** [comment]
there is a quite of refactoring so it won't be backported this time. The release is planned for Feb 2021
